kubernetes/kubernetes.go

```diff
@@ -4,33 +4,27 @@ import (
 	"context"
 	"fmt"
 	"net/http"
-	"os"
 
 	"github.com/rs/zerolog/log"
 	"sigs.k8s.io/controller-runtime/pkg/client"
 
 	"github.com/go-logr/zerologr"
 	ck "k8s.io/client-go/kubernetes"
 	restclient "k8s.io/client-go/rest"
-	"k8s.io/client-go/tools/clientcmd"
+	ctrl "sigs.k8s.io/controller-runtime"
 	rtlog "sigs.k8s.io/controller-runtime/pkg/log"
 )
 
 func CreateK8SClient(impersonate string) (*ck.Clientset, client.Client, error) {
 	var err error
 	var conf *restclient.Config
 	rtlog.SetLogger(zerologr.New(&log.Logger))
-	conf, err = restclient.InClusterConfig()
+
+	conf, err = ctrl.GetConfig()
 	if err != nil {
-		log.Debug().Err(err).Msg("")
-		log.Info().Msg("using out-of-cluster config")
-		conf, err = clientcmd.BuildConfigFromFlags("", os.Getenv("HOME")+"/.kube/config")
-		if err != nil {
-			return nil, nil, err
-		}
-	} else {
-		log.Info().Msg("using in-cluster config")
+		return nil, nil, err
 	}
```

Comment:
Does this do the same thing? How will the user know which kind of config is in use?

Reply:
Yes, the controller utils have the same semantic. Looking for `--kubeconfig, `.kube/config`, `KUBECONFIG` or in-cluster - that is what, for instance, the kube-builder framework is using. So by doing this we are sure to be aligned with most other things running inside of the cluster.

Reply:
It is this one: https://github.com/kubernetes-sigs/controller-runtime/blob/main/pkg/client/config/config.go#L76
but you are right it will not log whether you run "in-cluster" or not